### PR TITLE
Make sure `C_InvoiceLine.Price_UOM_ID` and `C_UOM_ID` are always set

### PR DESCRIPTION
**Pull request by TheBestPessimist**
https://github.com/metasfresh/metasfresh/issues/6458

### CODE REVIEW

File: de.metas.adempiere.adempiere/migration/src/main/sql/postgresql/system/10-de.metas.adempiere/5556570_sys_gh6458_Fix_C_InvoiceLines_With_Invalid_C_Uom_ID_or_Price_Uom_ID.sql [hunk 1]
```diff
@@ -0,0 +1,9 @@
+UPDATE c_invoiceline il
```
Comment (metas-ts) [inline de.metas.adempiere.adempiere/migration/src/main/sql/postgresql/system/10-de.metas.adempiere/5556570_sys_gh6458_Fix_C_InvoiceLines_With_Invalid_C_Uom_ID_or_Price_Uom_ID.sql:1]: pls add UpdatedBy and Updated with a "hardcoded" timestamp, e.g. `2020-04-09 13:23:07.035603+02`
on some DBs there might be "a lot" of C_InvoiceLines affected by this SQL

File: de.metas.swat/de.metas.swat.base/src/main/java/org/adempiere/invoice/service/impl/AbstractInvoiceBL.java [hunk 2]
```diff
@@ -674,6 +678,29 @@ public void updateDescriptionFromDocTypeTargetId(final org.compiere.model.I_C_In
...
+		final IInvoiceDAO invoiceDAO = Services.get(IInvoiceDAO.class);
+		final IProductBL productBL = Services.get(IProductBL.class);
+
+		final List<I_C_InvoiceLine> lines = invoiceDAO.retrieveLines(invoiceId); // // TODO tbp: create this method!
+		for (final I_C_InvoiceLine line : lines)
+		{
+			if (line.getC_UOM_ID() < 1)
+			{
+				final ProductId productId = ProductId.ofRepoId(line.getM_Product_ID());
+				final UomId stockUOMId = productBL.getStockUOMId(productId);
```
Comment (metas-ts) [inline de.metas.swat/de.metas.swat.base/src/main/java/org/adempiere/invoice/service/impl/AbstractInvoiceBL.java:693]: ok - getStockUOMId never returns null

File: de.metas.adempiere.adempiere/migration/src/main/sql/postgresql/system/10-de.metas.adempiere/5556570_sys_gh6458_Fix_C_InvoiceLines_With_Invalid_C_Uom_ID_or_Price_Uom_ID.sql (hunk 1 above)
Comment (TheBestPessimist) [reply de.metas.adempiere.adempiere/migration/src/main/sql/postgresql/system/10-de.metas.adempiere/5556570_sys_gh6458_Fix_C_InvoiceLines_With_Invalid_C_Uom_ID_or_Price_Uom_ID.sql:1]: done

File: de.metas.swat/de.metas.swat.base/src/main/java/org/adempiere/invoice/service/impl/AbstractInvoiceBL.java (hunk 2 above)
Comment (TheBestPessimist) [reply de.metas.swat/de.metas.swat.base/src/main/java/org/adempiere/invoice/service/impl/AbstractInvoiceBL.java:693]: Added `@NonNull` to make that explicit

**Pull request merged**